lib/tmpdir.rb

```diff
@@ -19,10 +19,11 @@ class Dir
   # Returns the operating system's temporary file path.
 
   def self.tmpdir
-    ['TMPDIR', 'TMP', 'TEMP', ['system temporary path', @@systmpdir], ['/tmp']*2, ['.']*2].find do |name, dir|
-      unless dir
-        next if !(dir = ENV[name]) or dir.empty?
-      end
+    candidate_dirs = ['TMPDIR', 'TMP', 'TEMP'].map(&ENV.method(:[]))
```

Comment:
This always looks up 3 environment variables, even unused.

Reply:
It's no longer needed: it's now iterating over a list of 6 directory paths, whereas previously it was iterating over a combination of 3 environment variable names _(`TMPDIR`, `TMP` and `TEMP`, which got turned into a directory path inside the block)_ and 3 directory paths _(`@@systmpdir`, `/tmp`, and `.`)_

For the first three entries, the `dir` in `(name, dir)` was always empty, as it was determined inside the block
For the last three entries, the `name` in `(name, dir)` was superfluous and totally unused.

This new version avoids all that confusion: it's now iterating over a homogeneous collection of 6 actual directory paths.

Reply:
It is still needed for the warning messages.

Reply:
Argh, I see what you mean... I _(wrongly!)_ assumed, because all tests passed, it was a kosher refactor, which clearly it isn't! :facepalm:

Back to the drawing board!  :sweat_smile: